### PR TITLE
feat(color, color-hex-input, color-swatch): add support for no-color

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #917

## Summary

This enhances `calcite-color` to allow it to be unset by using the `allowEmpty` property. 

When `allowEmpty` is enabled, users can clear the color by either setting the `value` attribute_ as `""` OR by setting the `value` _property_ as `null`. Also, when working with hex/RGB/HSV inputs, the color can be cleared by clearing an input and committing (Enter or blurring the input).

Additional notes:

### `calcite-color`

* supports `null` as a value (toggled via `allowEmpty`)
* backdoor `color` prop will also be `null` when empty
* value nudging is not allowed when empty
* saving/removing when empty is now allowed
* hex, RGB/HSV inputs will display empty when there is no color set
* thumbs from color field and hue slider will not be rendered until a new value is selected

### `calcite-color-hex-input`

* supports `null` as a value (toggled via `allowEmpty`)

## Preview

color-empty

**Note**: color swatch no longer renders when there is no color.

## Questions

* ~Is `allowEmpty` a good name for this behavior? When I write code or doc to describe this, I tend to use 'no color'. Originally, the prop was named `allowNoColor`, but I thought that may read as do not allow any colors.~ Proceeding w/ `allowEmpty`.

cc @bstifle @pemberdom @subgan82

### CODE REVIEW

File: src/components/calcite-color-hex-input/calcite-color-hex-input.tsx [hunk 1]
```diff
@@ -113,45 +147,60 @@ export class CalciteColorHexInput {
...
+    const inputValue = node.value;
+    const hex = `#${inputValue}`;
+    const willClearValue = this.allowEmpty && !inputValue;
 
-    if (isValidHex(hex) && isLonghandHex(hex)) {
+    if (willClearValue || (isValidHex(hex) && isLonghandHex(hex))) {
       return;
     }
 
+    // TODO: need to handle case where !valid hex and internal color is previously null - add test for this too!
```
Comment (driskull) [inline src/components/calcite-color-hex-input/calcite-color-hex-input.tsx:158]: Is there an issue for this part or happening in this PR?

**Review by driskull — APPROVED** [review]
👍🏻

File: src/components/calcite-color-hex-input/calcite-color-hex-input.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/calcite-color-hex-input/calcite-color-hex-input.tsx:158]: Whoops! 😅 Good catch. It'll happen in this PR.

**Comment by pemberdom** [comment]
>     Is allowEmpty a good name for this behavior? When I write code or doc to describe this, I tend to use 'no color'. Originally, the prop was named allowNoColor, but I thought that may read as do not allow any colors.

`allowEmpty` 👍

**Comment by driskull** [comment]
maybe `colorless`, `uncolored`??

**Comment by paulcpederson** [comment]
agree `allow-empty` is a good name for this prop. 

I'm not sure the `X` in the input is totally necessary in this state. Probably just omitting the color preview dot there would be sufficient, but I'll defer to @bstifle

**Comment by macandcheese** [comment]
Yeah I agree with maybe leaving out the X. It looks like a close or clear button. Probably not rendering anything at all makes sense IMO?

File: src/components/calcite-color-hex-input/calcite-color-hex-input.tsx (hunk 1 above)
Comment (jcfranco) [reply src/components/calcite-color-hex-input/calcite-color-hex-input.tsx]: 💥

**Comment by jcfranco** [comment]
WRT ⓧ: Noted. I'll wait for @bstifle to chime in. If we decide to not show a preview for the empty state, I'll revert the swatch changes for this since it'd be weird to have it not render when colorless.

**Comment by bstifle** [comment]
Lets just not show any circle or X when no fill. nothing there makes more sense to me

**Pull request merged**